### DISCUSSION TITLE
Issue with downloading History

### DISCUSSION THREAD

**Discussion by pandamega** [nianeyna/ao3downloader#156] [Q&A]
This is an amazing resource, and I have been able to fix most of my issues so far with the troubleshooting guide. I have unfortunately encountered a new issue that I'm not sure how can be resolved.

When downloading the full history page ~~there is an issue that every time the downloader accesses a fic to download, that fic gets added to the first page and adds more pages to the history indefinitely.~~ There is a significant lag between when a fic is accessed and when it is added to the history page so that several pages can be downloaded before the page numbers increase. However, this poses quite an issue: if you download pages 1-5 before the history catches up, the fics you just downloaded will be added to the history, so perhaps pages 1-3 are now fics you've just downloaded and the page 6 you were about to download next is now bumped down to page 9. 

When I started downloading my history I had under 900 pages, and now I am currently on page 1087 with nearly 200 pages to go... I think it might eventually catch up, but I'm worried those last pages might never be downloaded at this rate, and I'm worried some fics might have been skipped in the shuffle. it also tends to stop periodically and you have to make it start again using a new link with the page where it stopped (can't use the option to continue where it left off).

I also encountered another issue but I was able to resolve it, I just thought I'd let you know here in case you wanted to add it to the FAQ. If unfortunately an entire page has all the works deleted (happened in both my bookmarks and history) the downloader will stop and say "d to display menu again." The way around this is to either delete the deleted works form your history/bookmarks until there are a couple actual works on the page, or to start downloading again using a link for the next page that has works in it. 

**UPDATE:** So I've just realized that fics that you access again get _shuffled_ to the top of the history without adding a new entry, meaning the new pages are not coming from an infinite cycle of duplicate works. The reason my page numbers were increasing was because I used the option to download all works in a series which added more works to my history. this does have the issue of pushing back works in the download queue as the newly downloaded fics which I never accessed before are added to history, however the new pages added are not infinite and is it possible for the downloader to catch up! the downloader does periodically stop every now and then, sometimes for no reason I could fathom, but usually because the page was all deleted works. This happens periodically because the deleted works can no longer be accessed, leaving them to accumulate as the other fics are downloaded and moved to the top of the history, and when those accumulated deleted fics get pushed back into the queue when the history catches up to the newly added fics (from series) the program will stop when it hits a page full of deleted fics because there is nothing for it to download. Just restart the download from the next page that has fics. You'll know you've downloaded everything in your history if the last fics in your history are all deleted works. The last 20 pages or so of my history is now just deleted and hidden works. This whole thing shouldn't be an issue if you chose not to download all the works in a series.

So basically this isn't a problem with the program at all, but might be a bit difficult for someone using the downloader this way.
